### PR TITLE
20221219-no-sha-1-all-crypto

### PR DESCRIPTION
**Pull request by douzzer**
minor fixes to accommodate `--disable-sha` in combination with `--enable-all-crypto`.

tested with the incumbent tests, and new tests developed in parallel:

`wolfssl-multi-test.sh ... super-quick-check '.*no-sha-1.*' '.*140-3-dev.*'`

the new tests are:

```
fips-140-3-dev-optest-acvp-sp-asm-gcc-latest-no-sha-1
fips-140-3-dev-defaults-no-sha-1
fips-140-3-dev-all-crypto-no-sha-1
cppcheck-all-no-sha-1
clang-tidy-fips-140-3-dev-defaults-no-sha-1
clang-tidy-fips-140-3-dev-all-crypto-no-sha-1
clang-tidy-all-crypto-no-sha-1
sanitizer-all-crypto-no-sha-1
```

### CODE REVIEW

File: src/conf.c [hunk 1]
```diff
@@ -330,27 +330,30 @@ static unsigned long wolfSSL_CONF_VALUE_hash(const WOLFSSL_CONF_VALUE *val)
...
 {
     unsigned long ret = 0;
-#ifndef NO_SHA
+#if !defined(NO_SHA)
     wc_Sha sha;
     int strLen;
     byte digest[WC_SHA_DIGEST_SIZE];
+#elif !defined(NO_SHA256)
+    wc_Sha256 sha;
+    int strLen;
```
Comment (SparkiDev) [inline src/conf.c:347]: strLen is common.
Pull it out.

File: tests/api.c [hunk 2]
```diff
@@ -20372,7 +20372,7 @@ static int test_wc_RsaPublicEncryptDecrypt_ex(void)
     /* Encrypt */
     if (ret == 0) {
         ret = wc_RsaPublicEncrypt_ex(in, inLen, cipher, cipherSz, &key, &rng,
-                WC_RSA_OAEP_PAD, WC_HASH_TYPE_SHA, WC_MGF1SHA1, NULL, 0);
+                WC_RSA_OAEP_PAD, WC_HASH_TYPE_SHA256, WC_MGF1SHA1, NULL, 0);
```
Comment (SparkiDev) [inline tests/api.c:20375]: Test case checks for NO_SHA (about 20337) - change it to NO_SHA256.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: tests/api.c (hunk 2 above)
Comment (douzzer) [reply tests/api.c:20375]: good catch.  also fixed `WC_MGF1SHA1`.

**Comment by douzzer** [comment]
btw an example config for testing this:
```
--enable-all-crypto --disable-sha --disable-dsa --disable-crl --disable-ocspstapling
```
The last two options are tactical disables to work around lingering dependencies.  `make all` succeeds with or without those, but `make check` fails.

There are also still lingering pkcs7 dependencies, and IIRC a few other odds and ends teased out by full `--enable-all`.  And I think there are several prebaked artifacts in `certs/` that use SHA1.

**Review by SparkiDev — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Pull request merged**